src/Enclave.Sdk/Exceptions/EnclaveApiException.cs

```diff
@@ -20,8 +20,8 @@ public class ProblemDetailsException : Exception
     /// </summary>
     /// <param name="problemDetails">The problem details from the HTTP response.</param>
     /// <param name="response">The HTTP Response message.</param>
-    public ProblemDetailsException(ProblemDetails problemDetails, HttpResponseMessage response)
-        : base(problemDetails?.Title)
+    public EnclaveApiException(ProblemDetails problemDetails, HttpResponseMessage response, string? message = null)
+        : base(message ?? $"{problemDetails?.Title} - Check Problem Details for more details")
```

Comment:
It's not unusual to invoke a private static method call here in the call to base to do the computation you currently have to do in the `ProblemDetailsHttpMessageHandler`.  That way you can ditch the additional message parameter and do everything in the exception class.